### PR TITLE
Allow access to review page to most reviewers while restricting actions

### LINKED ISSUE #13566: Reviewer cannot access review tools page for add-on without public or pending versions
### Describe the problem and steps to reproduce it:

An add-ons reviewer (with permissions `Addons:PostReview`) cannot access review pages for add-ons where all files are disabled.

### What happened?

"You're not allowed to view this page"

### What did you expect to happen?

They should be allowed to view the page and also do write operations like sending a reviewer reply or commenting.

### Anything else we should know?
(Please include a link to the page, screenshots and any relevant files.)

### PR DESCRIPTION
**Pull request by diox**
This kinda reverses the previous behavior, allowing users with any kind of reviewer permission to see listed review pages, but restrict the actions depending on the permissions they have.

Unlisted continue to be restricted to unlisted reviewers.

Fixes #13566 (and more)

### CODE REVIEW

File: src/olympia/reviewers/views.py [hunk 1]
```diff
@@ -607,10 +607,9 @@ def queue_moderated(request):
 
 
 @any_reviewer_required
-@post_required
```
Comment (diox) [inline src/olympia/reviewers/views.py]: This is to make it work with `ReviewerTools:View`, aligning it with the rest of the views. It's a fairly minor thing, I almost dropped it (it's buried in the search form, when you add an app as a criteria)

File: src/olympia/access/acl.py [hunk 2]
```diff
@@ -151,10 +151,11 @@ def is_reviewer(request, addon):
     return check_addons_reviewer(request)
 
 
-def is_user_any_kind_of_reviewer(user):
+def is_user_any_kind_of_reviewer(user, allow_viewers=False):
```
Comment (willdurand) [inline src/olympia/access/acl.py]: So how is this supposed to work? I don't see this argument referenced in the body of the function.

Comment (willdurand) [reply src/olympia/access/acl.py]: Shouldn't we have new test cases for this change as well?

Comment (diox) [reply src/olympia/access/acl.py]: Yeah I missed a bit. I'll implement that change and test for it.

Comment (diox) [reply src/olympia/access/acl.py]: https://github.com/mozilla/addons-server/pull/13720/commits/a36a64bdac981e4da8a8c17ab043c49ce906aaac

**Pull request merged**